src/MongoDBEventStore.php

```diff
@@ -1,85 +1,131 @@
+
+
+    public function __construct(
+        Collection $eventCollection,
+        Serializer $payloadSerializer,
+        Serializer $metadataSerializer
+    ) {
+        $this->eventCollection = $eventCollection;
         $this->payloadSerializer = $payloadSerializer;
         $this->metadataSerializer = $metadataSerializer;
-
-        $this->collection = $mongodb->selectCollectionFromDefaultDatabase($collectionName);
     }
-
+    
     /**
      * {@inheritdoc}
      */
     public function load($id)
     {
-        $cursor = $this->collection->find(['uuid' => (string) $id]);
+        $cursor = $this->eventCollection
+            ->find([
+                'uuid' => (string) $id,
+            ],['sort' => ['payhead' => MongoCollection::ASCENDING]]);
 
-        $messages = [];
-        foreach ($cursor as $message) {
-            $messages[] = $this->denormalizeDomainMessage($message);
+        $events = [];
+
+        foreach ($cursor as $event) {
+            $events[] = $this->deserializeEvent($event);
         }
 
-        if (empty($messages)) {
+        if (!$events) {
             throw new EventStreamNotFoundException(sprintf('EventStream not found for aggregate with id %s', (string) $id));
         }
 
-        return new DomainEventStream($messages);
+        return new DomainEventStream($events);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function append($id, DomainEventStreamInterface $eventStream)
+    public function loadFromPlayhead($id, $playhead)
+    {
+        $cursor = $this->eventCollection
+            ->find([
+                'uuid' => (string) $id,
+                'playhead' => ['$gte' => $playhead]
+            ],['sort' => ['payhead' => MongoCollection::ASCENDING]]);
+
+        $events = [];
+        foreach ($cursor as $event) {
+            $events[] = $this->deserializeEvent($event);
+        }
+
+        return new DomainEventStream($events);
+    }
+
+    /**
+     * @param BSONDocument $event
+     *
+     * @return DomainMessage
+     */
+    private function deserializeEvent(BSONDocument $event)
+    {
+        return new DomainMessage(
+            $event['uuid'],
+            $event['playhead'],
+            $this->metadataSerializer->deserialize(json_decode($event['metadata'], true)),
+            $this->payloadSerializer->deserialize(json_decode($event['payload'], true)),
+            DateTime::fromString($event['recorded_on'])
+        );
+    }
+
+
+    /**
+     * {@inheritdoc}
+     */
+    public function append($id, DomainEventStream $eventStream)
     {
         $messages = [];
 
         foreach ($eventStream as $message) {
             $messages[] = $this->normalizeDomainMessage($message);
         }
 
-        $this->collection->insertMany($messages);
+        try {
+            $this->eventCollection->insertMany($messages);
+        } catch (BulkWriteException $exception) {
+            throw new DuplicatePlayheadException($eventStream, $exception);
```

Comment:
What about error code 11000? https://docs.mongodb.com/manual/core/index-unique/#unique-index-and-missing-field